lib/experimental-i18n-theme.json

```diff
@@ -1,32 +1,30 @@
 {
-	"settings": {
-		"typography": {
-			"fontSizes": [
-				"name"
-			],
-			"fontStyles": [
-				"name"
-			],
-			"fontWeights": [
-				"name"
-			],
-			"fontFamilies": [
-				"name"
-			],
-			"textTransforms": [
-				"name"
-			],
-			"textDecorations": [
-				"name"
-			]
-		},
-		"color": {
-			"palette": [
-				"name"
-			],
-			"gradients": [
-				"name"
-			]
-		}
+	"typography": {
```

Comment:
@gziolo @swissspidy This PR is going to change the shape of theme.json to make settings & styles top-level keys. In the future, we may have a theme.json along the lines of:

```json
{
  "name": "TwentyTwentyOne",
  "description": "...",
  "customTemplates": ...,
  "textDomain": ...,
  "version": 1,
  "settings": {
    "global": { ... },
    "core/paragraph": { ... }
  },
  "styles": {
    "global": { ... },
    "core/paragraph": { ... }
  }
}
```

At the moment we only need to translate presets (that are under a block selector, e.g.: `settings.global.color.palette`. However, given all the new planned features to be potentially merged in theme.json I am uncertain as to whether we'll need to translate more things in the future ― I'd think that we're going to need to translate more things.

Given this context, I wanted to ask what do you think would be a future-proof structure for the `i18n-theme.json`?

Reply:
I don't think it changes that much for the current implementation we have. As far as I can tell it only swaps the order for two first keys for every translation path listed. 

I personally would like yo exercise the idea of using JSON schema to include an additional property for translatable strings (schema allows them):
- `translateContext` (`string`)

REST API defines schemas already:

https://developer.wordpress.org/rest-api/extending-the-rest-api/schema/

In general, such schema has many benefits as it allows to auto-generate documentation for `theme.json` and `block.json`. You can provide tools for validation (e.g. webpack does it). In the case of translations my only concern would be how to parse that schema in the WordPress core. The structure proposed by Jorge is way more efficient 😃 Unless we have both the schema and simplified version for what `theme.json` loader and WP-CLI can consume.

Reply:
I'm not familiar with the JSON schema but happy to hear thoughts on that.

I've done some digging and found two proposals: [this for declaring shape](https://github.com/WordPress/gutenberg/pull/27380#discussion_r540308220) and [this for adding contexts for translation](https://github.com/WordPress/gutenberg/pull/28246#issuecomment-761117898). Wasn't involved a lot in this topic, so, please, share if I'm missing something.

Using those two things as a starting point, here's the result of merging and adapted them to the new shape (also added other keys to see how this could be extended):

```
{
    "futureKeyToTranslate": "Context for this other key",
    "settings": {
        "*": {
            "typography": {
                "fontSizes": {
                    '*': [ { "name": "Font size name" } ]
                }
            }
        }
    }
}
```

As far as I understand, `*` is used as a placeholder to indicate that there's a list of things here.

How does this feel? Is there anything missing we need to incorporate?

Trying to at least understand what's the upcoming shape for this so I can make the necessary adaptions to this PR.

Reply:
You only need to update https://github.com/WordPress/gutenberg/blob/master/lib/experimental-i18n-theme.json. Swap keys at the root of the file.

Reply:
did it like https://github.com/WordPress/gutenberg/pull/28110#discussion_r563171543 so marking as resolved